pkg/cgroup/linux.go

```diff
@@ -0,0 +1,213 @@
+// +build linux
+
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package cgroup implements parsing for all the cgroup
+// categories and functionality in a simple way.
+package cgroup
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// DO NOT EDIT following constants are chosen defaults for any kernel
+// after 3.x, please open a github issue https://github.com/minio/minio/issues
+// and discuss first if you wish to change this.
+const (
+	// Default value for any error or if cgroup is not set.
+	// this is the defautl if cgroups are not used/configured.
+	unlimitedMemoryLimit = 9223372036854771712
+
+	// Default string for looking for kernel memory param.
+	memoryLimitKernelParam = "memory.limit_in_bytes"
+
+	// Points to sys path memory path.
+	cgroupMemSysPath = "/sys/fs/cgroup/memory"
+
+	// Default docker prefix.
+	dockerPrefixName = "/docker/"
+
+	// Proc controller group path.
+	procCGroupPath = "/proc/%d/cgroup"
+)
+
+// CGEntries - represents all the entries in a process cgroup file
+// at /proc/<pid>/cgroup.
+type CGEntries map[string][]string
+
+// GetCGPathByName - returns the path matching the controller name.
+func (cg CGEntries) GetCGPathByName(cname string) (path string) {
```

Comment:
I think this function is not necessary. As map is its base type.  The consumer can directly do `cg[key]`, its up to the consumer what to do with value as string slice.

In case of `GetCGroupMemoryLimit()`, it could error out if more than one value in slice as we agreed previously.